consensus/src/processes/pruning_proof/mod.rs

```diff
@@ -88,15 +100,59 @@ impl<T> Clone for CachedPruningPointData<T> {
     }
 }
 
+struct TempProofContext {
+    headers_store: Arc<DbHeadersStore>,
+    ghostdag_stores: Vec<Arc<DbGhostdagStore>>,
+    relations_stores: Vec<DbRelationsStore>,
+    reachability_stores: Vec<Arc<parking_lot::lock_api::RwLock<parking_lot::RawRwLock, DbReachabilityStore>>>,
+    ghostdag_managers:
+        Vec<GhostdagManager<DbGhostdagStore, DbRelationsStore, MTReachabilityService<DbReachabilityStore>, DbHeadersStore>>,
+    db_lifetime: DbLifetime,
+}
+
+#[derive(Clone)]
+struct RelationsStoreInFutureOfRoot<T: RelationsStoreReader, U: ReachabilityService> {
+    relations_store: T,
+    reachability_service: U,
+    root: Hash,
+}
+
+impl<T: RelationsStoreReader, U: ReachabilityService> RelationsStoreReader for RelationsStoreInFutureOfRoot<T, U> {
```

Comment:
A more fancy approach would (move out of this file and) do a composition where filtering  can be applied in layers and then using extensions you would do something like 

```
filtered_relations = base_relations
    .intersect_with_future_of(root, reachability_service)
    .intersect_with_past_of(selected_tip, reachability_service) // some places in the code below do this step as well
``` 

Can be done in a style like `UtxoView` instances can be composed. Probably better if done in a separate PR

Reply:
Created an issue to track it in: https://github.com/kaspanet/rusty-kaspa/issues/585